extensions/interactions/LogicProof/static/js/student.js

```diff
@@ -994,12 +994,16 @@ var logicProofStudent = (function() {
   *          }
   */
   var buildProof = function(proofString, questionInstance) {
-    if (proofString.slice(-1) === '\n') {
-      proofString = proofString.slice(0, proofString.length - 1);
-    }
     var lineStrings = proofString.split('\n');
+    // Ignore blank lines at the end

```

Comment:
From a design perspective, I'm afraid I don't understand the rationale for handling this as a courtesy, but not handling other similar cases. Unless commenting on extra spaces at the beginning of lines, or newlines in the middle, has some sort of pedagogical value in learning how to write logical proofs, I'd strongly recommend allowing it and normalizing as needed.

In general, I want to advocate being less restrictive about the input we receive; this is a point I learned from our studies with users. Having to deal with nits that are not directly related to what they are trying to learn can easily lead to frustration on the part of the learner. So, I think it's great that you're trimming whitespace at the end, but it would be nice to handle similar things throughout as well.